### PR TITLE
Ch3 fixes

### LINKED ISSUE #244: Collection of basic Chapter 3 issues
- [x] Blockades
- [x] Required items (Daedalus?)
- [x] Make sure you can't die (can happen relative easily with big stack especially)
- [ ] Creeps can be killed before goal is ready (?)
- [x] Riki needs to be spawned and handled correctly
- [x] Spawn ODPixel
- [x] Highlighting

### PR DESCRIPTION
**Pull request by Wouterz90**
#244

### CODE REVIEW

**Review by RobinKa — CHANGES_REQUESTED** [review]
Please use proper formatting, it's very hard to read and undoing it every time takes a lot of time. (especially for the function calls)

**Comment by Wouterz90** [comment]
I have no idea, just using the alt shift f command which seems to cut of lines very soon, opening another editor to format now

**Comment by RobinKa** [comment]
Still a lot of changed stuff. Is it possible you start from how it was and only make the actual changes from there?

**Comment by Wouterz90** [comment]
I guess that's the best solution for now

**Comment by RobinKa** [comment]
image

I received 2 of each neutral item

**Comment by RobinKa** [comment]
- When you fail to stack at the 0/5 part, ODPixel says his line twice
- Camp after the N/5 section doesn't respawn and goal immediately completes, getting the player stuck

image

**Comment by Wouterz90** [comment]
Ill test this more thoroughly tomorrow

Should have solved your second point, my solution for the first wasn't right. I got an idea for that.

**Comment by Wouterz90** [comment]
Need to remove 1 more line

File: game/scripts/vscripts/Sections/Chapter3/Chapter3.ts [hunk 1]
```diff
@@ -208,38 +248,44 @@ const onStart = (complete: () => void) => {
...
-
-                    GetUnitsInsidePolygon(creepCampBox).forEach(unit => {
-                        if (!creepArr.includes(unit)) {
-                            creepArr.push(unit);
-                        }
-                    });
-
                     tryCount++;
                 });
+                //playerHero.AddNewModifier(undefined, undefined, "modifier_keep_hero_alive", undefined);
```
Comment (RobinKa) [inline game/scripts/vscripts/Sections/Chapter3/Chapter3.ts:258]: Remove this?

File: game/scripts/vscripts/Sections/Chapter3/Chapter3.ts [hunk 2]
```diff
@@ -208,38 +248,44 @@ const onStart = (complete: () => void) => {
...
                 playerHero.AddNewModifier(undefined, undefined, "modifier_deal_no_damage", undefined);
+                let odPixel = ctx[CustomNpcKeys.ODPixelMudGolem] as CDOTA_BaseNPC;
+                odPixel.SetAbsOrigin(GetGroundPosition(markerLocation, undefined));
+                setUnitPacifist(odPixel, true);
             }),
-            tg.audioDialog(LocalizationKey.Script_3_Opening_19, LocalizationKey.Script_3_Opening_19, ctx => ctx[CustomNpcKeys.SunsFanMudGolem]),
-            tg.loop(_ => tryCount < 5, _ => {
+            tg.audioDialog(LocalizationKey.Script_3_Opening_19, LocalizationKey.Script_3_Opening_19, ctx => ctx[CustomNpcKeys.ODPixelMudGolem]),
+            tg.loop(_ => tryCount <= 5, _ => { // Set this back to 5 // Remove the hardcoding
```
Comment (RobinKa) [inline game/scripts/vscripts/Sections/Chapter3/Chapter3.ts:265]: Comment still accurate / remove it?

**Comment by RobinKa** [comment]
- After skip neutral items stick around (need to remove those in setupState too)
- Item slot if hardcoded for neutrals
- OD should say his stack unsuccessful line instead of repeating the previous one on stack fail
- Sometimes creeps randomly respawn (from ingame time I assume) which is kinda confusing

image

**Pull request merged**